### ISSUE TITLE
Phase 10 readiness: fp16 quantisation + browser-runtime parity check

### ISSUE THREAD

**Issue by khawkins98** [khawkins98/foveacast-training#15]
Spun out of the DevRel review of PR #14 (Phase 8 ONNX export). Both items are Phase 10-blocking but Phase 8 scope-out — Phase 8's gate is PyTorch↔onnxruntime CPU parity, which #14 closed. Before the `.onnx` actually ships into Foveacast's browser path, these two need addressing.

## A. fp16 quantisation for the release artefact

Current `releases/foveacast-stock-dev.onnx` is 106.4 MB — MSI-Net's 25M fp32 params × 4 bytes. V2's UNISAL was 12.5 MB for reference. 100 MB is blocker-grade for a browser download.

fp16 halves it to ~53 MB; likely good enough without going to int8 quantisation (which is a separate toolchain). Approach:

```python
# After torch.onnx.export, before onnx.save_model:
from onnxconverter_common.float16 import convert_float_to_float16
model_proto = convert_float_to_float16(model_proto, keep_io_types=True)
```

**Gotcha:** `_normalize` in `msinet.py` divides by `(eps + max_per_image)` with `eps=1e-7`. fp16 min normal is ~6e-5; `1e-7` underflows to zero. Two options:
- Bump `eps` to `1e-5` everywhere in `_normalize` and `_tf1_bilinear_upsample`. Affects PyTorch numerics slightly — re-run parity + Phase 6 eval to confirm no regression.
- Keep I/O as fp32 and the normaliser as fp32 while quantising only the conv weights. `onnxconverter_common.float16` has keep-list options for this.

The second is cleaner — only the 25M conv weights go fp16 (12.5 MB saving × ~8 = 50 MB), and the tiny eps arithmetic stays fp32.

## B. onnxruntime-web parity check (WASM + WebGPU EPs)

Python `onnxruntime` CPU parity does NOT prove browser parity. ORT-web has three execution providers with different op coverage:

- **WASM** — most permissive; most ops supported. Likely-safe for our export.
- **WebGPU** — thinner op/shape-broadcast surface. `_tf1_bilinear_upsample`'s rank-4 gather with broadcast-expanded 1-D indices on two axes simultaneously is historically a sharp edge.
- **WebGL** — similar to WebGPU. Foveacast V2 UNISAL currently uses... (need to check; Phase 10 plumbing).

**Task:** run the exported `.onnx` in a minimal Node/browser test harness with the same EP(s) Foveacast intends to use. Compare outputs to PyTorch's. Any discrepancy there is what actually matters for V3 shipping, not the Python-side number.

Minimal harness could be a short JS file using `onnxruntime-web` npm package loading the `.onnx` + a fixture input tensor (e.g. the same fixed-seed one the Python parity uses). Could live at `scripts/browser_parity_check.js` or similar.

## C. Expanded parity trials (landed in #14 follow-up commit, not tracked here)

Reviewer also flagged that parity trials were too narrow (only 3 random bs=1 inputs). Fixed in a follow-up commit on the Phase 8 branch: added batch_size=2 trial and saturated (all-0, all-255) trials. 6 trials total, max abs err 2.32e-06 vs previous 1.79e-06. Not a separate issue item, but worth noting the fuller coverage against which fp16 and browser-runtime parity should be validated.

## Ordering and merge recommendation

A → B, since (B) needs a valid `.onnx` and (A) changes which `.onnx` we're validating. Both should land before any Phase 10 PR opens on Foveacast itself — the `.onnx` attached to the GitHub Release needs to be both small enough to load and provably-correct in the browser runtime Foveacast will use.

Scope estimate: A is half a day (~1 hour code + parity re-run + small CLI flag + doc updates). B is half a day (~2 hours stand up the JS harness + run + document + maybe a Playwright integration depending on how fancy we get).

Relates to #1 Phase 8 follow-up, Phase 10 prep. Not blocking PR #14 (Phase 8 gate is closed by the Python parity).
